src/ossia/protocols/midi/midi_protocol.cpp

```diff
@@ -69,19 +76,39 @@ bool midi_protocol::set_info(midi_info m)
 
     if (m_info.type == midi_info::Type::Input)
```

Comment:
I wonder if this function cannot be simplified a bit by having a :

     static std::string get_midi_port_name(ossia::net::device_base* dev, midi_settings ....) { ... } 

function which implements this if / else logic which is just here to choose a name, and then just have

```c++
auto& midi_impl =  m_info.type == Input ? m_input : m_output;
if(!isVirtual)
  midi_impl->open_port(m_info.port, get_midi_port_name(m_dev, m_info);
else
  midi_impl->open_virtual_port(m_info.port, get_midi_port_name(m_dev, m_info);
```

Reply:
added the proposed function

Reply:
perfect thanks !